backend/src/main/java/co/inventorsoft/scripty/service/ProjectService.java

```diff
@@ -0,0 +1,104 @@
+package co.inventorsoft.scripty.service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import co.inventorsoft.scripty.exception.ApplicationException;
+import co.inventorsoft.scripty.model.dto.ProjectDto;
+import co.inventorsoft.scripty.model.entity.Project;
+import co.inventorsoft.scripty.model.entity.User;
+import co.inventorsoft.scripty.repository.ProjectRepository;
+import co.inventorsoft.scripty.repository.UserRepository;
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+
+/**
+ * @author lzabidovsky 
+ */
+@Service
+@Transactional
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class ProjectService {
+
+	@Autowired
+	ProjectRepository projectRepository;
+
+	@Autowired
+	UserRepository userRepository;
+
+	final static String DEFAULT_DESCRIPTION = "";
+	final static boolean DEFAULT_VISIBILITY = true;
+
+	@Value("${path.local.repo}")
+	String pathLocalRepo;
+	@Value("${directory.separator}")
+	String directorySeparator;
+
+	public long saveProject(ProjectDto project, String username) {
+		User user = userRepository.findByEmail(username).get();
```

Comment:
You don't need `Optional` when you don't use it properly. In this case method `get()` with throw `NullPointerException` in case user with specified email is not found. Usage of method `Optional.get()` is considered a bad practice. Ideally this method shouldn't even exist in this class